### ISSUE TITLE
[boost::boost] boost/1.71.0: python2 incompatibility

### ISSUE THREAD

**Issue by nilsnolde** [conan-io/conan-center-index#8416]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **boost/1.71.0**
  * Conan version: **conan 1.43.0**
  * Python version: **Python 2.7.13**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=6
compiler.libcxx=libstdc++
build_type=Release
```

### Steps to reproduce (Include if Applicable)

I'm not even sure if you consider this an issue. I was surprised to see that conan still supports python2. anyways, here it goes:

install boost (probably any version) via Python 2.7.x:

```
conanfile in /root/.conan/data/boost/1.71.0/_/_/export/conanfile.py
  File "/root/.conan/data/boost/1.71.0/_/_/export/conanfile.py", line 521
    tools.get(**self.conan_data["sources"][self.version],
                                                        ^
SyntaxError: invalid syntax
```

the problem seems that python2 doesn't support dict expansion (is that the right term?) to create dicts, e.g. `d2 = {**d1}`  is illegal, then this fails:
https://github.com/conan-io/conan-center-index/blob/94157ccd6f8ddeb5c586cd456acad5affb6f74f4/recipes/boost/all/conanfile.py#L520-L524

basically the first usage of `**dict` is invalid, the second is fine.

**Comment by SSE4** [comment]
Python 2.x is no longer supported: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-is-the-policy-for-supported-python-versions

**Comment by nilsnolde** [comment]
ah ok, guess I got confused bcs conan executable still supports python2. probably just for legacy reasons of alternative package indices then..
